wgpu-native/src/command/transfer.rs

```diff
@@ -81,6 +81,17 @@ pub fn command_encoder_copy_buffer_to_buffer<B: GfxBackend>(
     // borrow the buffer tracker mutably...
     let mut barriers = Vec::new();
 
+    let src_buffer_usage = buffer_guard[source].usage;
```

Comment:
similarly, let's use the `src_buffer.usage` and `dst_buffer.usage` when it's available, no need to rush for this check